2.0/m_antibotctcp.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *       Copyright (C) Nikos `UrL` Papakonstantinou <url.euro@gmail.com>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.      If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "inspircd.h"
+/* $ModConfig: <antibotctcp ctcp="VERSION" msgonreply="true" showlogs="true" accepted="Howdy buddy,you are authorized to use this server!"  declined="You have been blocked!Please get a better client."> */
+/* $ModDesc: Blocks clients not replying to CTCP like botnets/spambots/floodbots. */
+/* $ModAuthor: Nikos `UrL` Papakonstantinou */
+/* $ModAuthorMail: url@mirc.com.gr */
+/* $ModDepends: core 2.0 */
+
+/*
+ * Supports connect class param
+ * Add antibotctcp="true" which enables or disables the module on that port(s).
+ */
+
+class ModuleAntiBotCTCP : public Module
+{
+	LocalIntExt ext;
+	bool msgonreply;
+	bool showlogs;
+	std::string accepted;
+	std::string declined;
+	std::string ctcp;
+ public:
+	ModuleAntiBotCTCP()
+	: ext("ctcptime_wait", this)
+	{
+	}
+
+	void init()
+	{
+		ServerInstance->Modules->AddService(ext);
+		Implementation eventlist[] = { I_OnUserRegister, I_OnPreCommand, I_OnCheckReady, I_OnRehash };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
+		OnRehash(NULL);
+	}
+
+	Version GetVersion()
+	{
+		return Version("Blocks clients not replying to CTCP.");
+	}
+
+	void OnRehash(User* user)
+	{
+		ConfigTag* tag = ServerInstance->Config->ConfValue("antibotctcp");
+		ctcp = tag->getString("ctcp");
+		showlogs = tag->getBool("showlogs", true);
+		msgonreply = tag->getBool("msgonreply", true);
+		declined = tag->getString("declined");
+		accepted = tag->getString("accepted");
+	}
+
+	ModResult OnUserRegister(LocalUser* user)
+	{
+		ConfigTag* tag = user->MyClass->config;
+		if (tag->getBool("antibotctcp", true))
+		{
+			if (ctcp.empty())
```

Comment:
You can get rid of this and block and pass "VERSION" as the second parameter to getString("ctcp").